### ISSUE TITLE
orderly_artefact() requires description vs. orderly_resource() only file list

### ISSUE THREAD

**Issue by jeffeaton** [mrc-ide/orderly2#150]
On using orderly2 with R function declarations, the arguments feel somewhat incongruous:

* `orderly_artefact()` _requires_ a `description = ` argument
* `orderly_resource()` only accepts a list of files (no description attached)

Is there a conceptual reason why an artefact needs a description (rather than only requiring declaration of the files)?

Intuitive harmonised interface for me:
* `orderly_artefact()` requires `files = ` argument, `description = ` argument becomes an optional argument (ideally second)
* `orderly_resource()` gains an optional `description = ` argument
* `orderly_dependency()` gains an optional `description = ` argument

Maybe there's a reason that doesn't make sense

**Comment by richfitz** [comment]
No, this makes sense, and should only be minimally disruptive if we make it right now.  The reason for the existing behaviour was it mirrored the previous yaml most closely

**Comment by richfitz** [comment]
So, looking further - it will be easy enough to make the description of artefact optional, though this will require a period of adjustment before we can change the argument order, based on patterns of usage.

It's a bit harder to get descriptions added to resource and dependency though, because of the way these are stored in the final metadata.  We can change this, but it's a more disruptive change than I would like to do lightly, even if things are still new, partly because nobody previously asked for this in orderly1 (so it's not clear that it would warrant the disruption).  I'll get onto making this optional for artefacts though...

**Comment by plietar** [comment]
While we are at it, it might be worth changing `orderly_shared_resource` to take a list of files as well, instead of a `...`. This had come up when I was doing #136.
